spec/rubocop/cop/style/extend_self_spec.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::ExtendSelf, :config do
+  subject(:cop) { described_class.new(config) }
+
+  context 'when configured to enforce `module_function` style' do
+    let(:cop_config) do
+      {
+        'EnforcedStyle' => 'module_function'
+      }
+    end
+
+    it 'registers an offense when used alone in a module' do
+      expect_offense(<<-RUBY.strip_indent)
+        module Foo
+          extend self
+          ^^^^^^^^^^^ Use `module_function` instead of `extend self`.
+        end
+      RUBY
+    end
+
+    it 'registers an offense when used alone in a module wrapped in begin' do
+      expect_offense(<<-RUBY.strip_indent)
+        module Foo
+          begin
+            extend self
+            ^^^^^^^^^^^ Use `module_function` instead of `extend self`.
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense when used with other statements in a module' do
+      expect_offense(<<-RUBY.strip_indent)
+        module Foo
+          extend self
+          ^^^^^^^^^^^ Use `module_function` instead of `extend self`.
+
+          def bar; end
+        end
+      RUBY
+    end
+
+    it 'does not register an offense when used in a class nested in a module' do
+      expect_no_offenses(<<-RUBY.strip_indent)
+        module Foo
+          class Bar
+            extend self
+          end
+        end
+      RUBY
+    end
+
+    it 'auto-corrects `extend self` to `module_function`' do
+      new_source = autocorrect_source(<<-RUBY.strip_indent)
+        module Foo
+          extend self
+        end
+      RUBY
+
+      expected_source = <<-RUBY.strip_indent
+        module Foo
+          module_function
```

Comment:
My slight gripe with those examples are that this is not real code, without some method definition.

Reply:
Right. I will probably add both, since the AST is (sadly) different for a single expression vs. multiple.